src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -405,64 +423,83 @@ repl_req_ptr_t RaftReplDev::applier_create_req(repl_key const& rkey, sisl::blob
     // thing. So take state mutex and allocate the blk
     std::unique_lock< std::mutex > lg(rreq->state_mtx);
     if (rreq->state.load() & uint32_cast(repl_req_state_t::BLK_ALLOCATED)) { return rreq; }
+
+    auto hints_result = m_listener->get_blk_alloc_hints(user_header, data_size);
```

Comment:
are we using the `get_blk_alloc_hints` as pre-append?

This might not enough as we might want to reject based on business logical.  say if we get a putblob but the shard doesnt exists in the replica etc.

Reply:
yes，what`s more, the pre-append should also be called in leader, since leader might reject some request base on some logic(business or not)

Reply:
@xiaoxichen put_blob before shard create is already covered in the follower, because all of them goes through this one call `get_blk_alloc_hints` even for put_blob. We could change the api name to something with more meaning and put more info on it. Lets keep it open and evolve that API.

@JacksonYao287 As long as leader is the always the proposer, I am not sure if we have to do anything for the leader. After all why would leader propose something it needs to veto on?